hat/app/org/hatdex/hat/api/controllers/Authentication.scala

```diff
@@ -223,4 +225,100 @@ class Authentication @Inject() (
         Future.successful(Unauthorized(Json.toJson(ErrorMessage("Invalid Token", "Token does not exist"))))
     }
   }
+
+  /*
+    BEGIN: hat-claim
+    Entire Section below is on HAT Claim.
+    TODO: Refactor to New File?
+   */
+  /**
+   * Sends an email to the owner with a link to claim the hat
+   */
+  def claim(applicationId: String): Action[AnyContent] = SecuredAction(WithRole(Owner())).async { implicit request =>
+    val email = request.dynamicEnvironment.ownerEmail
+    val response = Ok(Json.toJson(SuccessResponse("You will shortly receive an email with claim instructions")))
+
+    applicationsService.applicationStatus().flatMap { applications =>
+      applications.find(_.application.id.equals(applicationId)) match {
+        case Some(app) =>
+          usersService.listUsers.map(_.find(_.roles.contains(Owner()))).map {
+            case Some(_) =>
+              val token = MailTokenUser(email, isSignUp = false)
+              tokenService.create(token).map { _ =>
+                val scheme = if (request.secure) {
+                  "https://"
+                }
+                else {
+                  "http://"
+                }
+                val claimLink = s"$scheme${request.host}/#/hat/claim/${token.id}"
+                mailer.claimHat(email, claimLink, app)
+              }
+              response
+            case None => response
+          }
+        //Future.successful(response)
+
+        /*usersService.listUsers.map(_.find(_.roles.contains(Owner()))).flatMap {
+            case Some(_) =>
+              println("here")
+              val token = MailTokenUser(email, isSignUp = false)
+              tokenService.create(token).map { _ =>
+                val scheme = if (request.secure) {
+                  "https://"
+                }
+                else {
+                  "http://"
+                }
+                val claimLink = s"$scheme${request.host}/#/hat/claim/${token.id}"
+                mailer.claimHat(email, claimLink, appOpt.get)
+                Future.successful(response)
+              }
+
+            case None => {
+              println("tere")
+              Future.successful(response)
+            }
+          }*/
+        case None => Future.successful(NotFound(Json.toJson(ErrorMessage("Application Not Found", "The application id is invalid."))))
```

Comment:
In the case the application ID is not found, the claim process should still proceed, but simply omit the parter logo from email.

Reply:
Fixed as requested.